core/includes/theme.inc

```diff
@@ -1951,7 +1951,19 @@ function theme_dropbutton_wrapper($variables) {
  */
 function theme_image($variables) {
   $attributes = $variables['attributes'];
-  $attributes['src'] = file_create_url($variables['uri']);
+
+  // Check for missing 'uri', allow D7's 'path' instead.
+  $uri = $variables['uri'];
+  if (!$uri && !empty($variables['path'])) {
+    watchdog('Image',
```

Comment:
No objections on lower-case for `$type`. 

The line was split, exactly as you say, for readability.  I don't mind setting it to be all on one line.

On the `watchdog()` parameters, I do see the source of your quote "if message is already translated..." in the [watchdog documentation](https://api.backdropcms.org/api/backdrop/core%21includes%21bootstrap.inc/function/watchdog/1), but there are two problems here with the core code. 

First, the message _can_ be translated, there just aren't any variable substitutions.  The French version might be "theme_image() appelée avec obsolète «path», utilsez «uri» plutôt."

Second, it may not matter, because I can't find where any translation is happening.  The `watchdog()` function just packs the message and variables into an array before invoking `hook_watchdog`. The two in-core logging modules, `dblog` and `syslog`, do very little with the variables.  `dblog` saves the message untouched in one DB field, and the variables serialized into another.  `syslog` use `strtr()` to do the substitutions, but never checks if the translation layer is available, let alone call `t()`.

Bonus issue: shouldn't the translation layer decide if something is translatable or not, not the caller?

From a practical standpoint, it doesn't matter if `array()` (the default in the function signature) or `NULL` is used.  I don't mind making it one or the other to satisfy a review, but I am curious what is the right thing to do give the above issues.